Sources/SwiftSyntax/SyntaxProtocol.swift

```diff
@@ -225,13 +225,10 @@ extension SyntaxProtocol {
   }
 
   public var keyPathInParent: AnyKeyPath? {
-    guard let parent = self.parent else {
-      return nil
-    }
-    guard case .layout(let childrenKeyPaths) = parent.kind.syntaxNodeType.structure else {
+    guard case .layout(let childrenKeyPaths) = self._syntaxNode.parent?.kind.syntaxNodeType.structure else {
```

Comment:
Is `self._syntaxNode.parent` faster than `parent`?

Reply:
That was my measurement, but I'm not 100% sure tbh. I will double check.

Reply:
Removed this commit from the PR for now